### PR TITLE
fix(image_projection_based_fusion): add copying input fields to output for segmentation_pointcloud_fusion

### PR DESCRIPTION
**Pull request by badai-nguyen**
## Description

- To add copying the input field to ouput to make it works properly in case of input pointcloud has other fiels (e.g intensity, channel..)
- Related to the issue: https://github.com/autowarefoundation/autoware.universe/issues/6785

## Tests performed

- Confirmed by logging_simulator that the output of segmentation_pointcloud_fusion node has intensity field as input.
Screenshot from 2024-06-03 09-57-18

Screenshot from 2024-06-03 09-57-33

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by YoshiRi — APPROVED** [review]
LGTM

File: perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp [hunk 1]
```diff
@@ -116,16 +131,21 @@ void SegmentPointCloudFusionNode::fuseOnSingleImage(
...
+        input_pointcloud_msg, point_step, global_offset, output_cloud, output_pointcloud_size);
     }
   }
 
-  pcl::toROSMsg(output_cloud, output_pointcloud_msg);
-  output_pointcloud_msg.header = input_pointcloud_msg.header;
+  output_cloud.data.resize(output_pointcloud_size);
+  output_cloud.row_step = output_pointcloud_size / output_cloud.height;
+  output_cloud.is_bigendian = input_pointcloud_msg.is_bigendian;
+  output_cloud.is_dense = input_pointcloud_msg.is_dense;
```
Comment (kminoda) [inline perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp:147]: Any specific reason why these two lines are declared nearby the following (previous) lines?
```cpp
  output_cloud.fields = input_pointcloud_msg.fields;
  output_cloud.header = input_pointcloud_msg.header;
  output_cloud.height = input_pointcloud_msg.height;
  output_cloud.point_step = input_pointcloud_msg.point_step;
```

File: perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp [hunk 2]
```diff
@@ -85,29 +85,44 @@ void SegmentPointCloudFusionNode::fuseOnSingleImage(
...
+  int y_offset = input_pointcloud_msg.fields[pcl::getFieldIndex(input_pointcloud_msg, "y")].offset;
+  int z_offset = input_pointcloud_msg.fields[pcl::getFieldIndex(input_pointcloud_msg, "z")].offset;
+  size_t output_pointcloud_size = 0;
+  output_cloud.data.clear();
+  output_cloud.data.resize(input_pointcloud_msg.data.size());
+  output_cloud.fields = input_pointcloud_msg.fields;
+  output_cloud.header = input_pointcloud_msg.header;
+  output_cloud.height = input_pointcloud_msg.height;
+  output_cloud.point_step = input_pointcloud_msg.point_step;
+  std::cout << "point_step: " << transformed_cloud.point_step << std::endl;
```
Comment (kminoda) [inline perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp:99]: Is this addition intended? (Or just a debug message which you've forgot to remove?)

**Review by kminoda — COMMENTED** [review]
Thanks for the PR!

File: perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp (hunk 2 above)
Comment (badai-nguyen) [reply perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp]: @kminoda Thanks for comment, It was my typo. I fixed https://github.com/autowarefoundation/autoware.universe/pull/7224/commits/b44464fa858a27542ec862b3aa5d798395db3625

File: perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp (hunk 1 above)
Comment (badai-nguyen) [reply perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp]: @kminoda No, so I changed as your suggestion. https://github.com/autowarefoundation/autoware.universe/pull/7224/commits/61a137dbe6417424cabaa5a5a986227afa9dd348

**Review by kminoda — APPROVED** [review]
LGTM, thanks!
